specification/modsimworkbench/resource-manager/Microsoft.ModSimWorkbench/preview/2021-03-01-preview/connectors.json

```diff
@@ -0,0 +1,909 @@
+          },
+          {
+            "in": "path",
+            "name": "resourceGroup",
+            "required": true,
+            "type": "string",
+            "description": "The name of the resource group"
+          },
+          {
+            "in": "path",
+            "name": "workbenchName",
+            "required": true,
+            "type": "string",
+            "description": "The name of the workbench"
+          },
+          {
+            "name": "chamberName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the chamber."
+          },
+          {
+            "in": "path",
+            "name": "name",
+            "required": true,
+            "type": "string",
+            "description": "The name of the connector."
+          },
+          {
+            "in": "body",
+            "name": "body",
+            "required": true,
+            "description": "License key parameters",
+            "schema": {
+              "$ref": "#/definitions/LicenseKeyParameters"
+            }
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Success"
+          },
+          "202": {
+            "description": "Accepted",
+            "headers": {
+              "Location": {
+                "type": "string"
+              }
+            }
+          },
+          "default": {
+            "description": "Default Error Response",
+            "schema": {
+              "$ref": "./modsimworkbenches.json#/definitions/ErrorResponse"
+            },
+            "x-ms-error-response": true
+          }
+        },
+        "x-ms-long-running-operation": true,
+        "x-ms-long-running-operation-options": {
+          "final-state-via": "location"
+        },
+        "x-ms-examples": {
+          "Connectors License Add": {
+            "$ref": "./examples/ConnectorsLicenseAdd.json"
+          }
+        },
+        "security": [
+          {
+            "azure_auth": [
+              "user_impersonation"
+            ]
+          }
+        ]
+      }
+    }
+  },
+  "definitions": {
+    "RemoteDesktop": {
+      "type": "object",
+      "description": "Remote Desktop properties.",
+      "properties": {
+        "etx": {
+          "$ref": "#/definitions/Etx",
+          "description": "The etx object."
+        }
+      }
+    },
+    "Etx": {
+      "type": "object",
+      "properties": {
+        "connectionNodeCount": {
+          "type": "integer",
+          "readOnly": true,
+          "format": "int32",
+          "description": "The connection node count."
+        },
+        "licenseActivated": {
+          "description": "The license activated.",
+          "type": "string",
+          "readOnly": true,
+          "enum": [
+            "True",
+            "False"
+          ],
```

Comment:
Having an enumeration with 'True' and 'False' values is an antipattern, there should be some kind of linter rule against this.

Reply:
Conside changign to something like 'Activated', 'Expired', and 'Invalid'?

Reply:
[Maybe we could make an exception in the case of query parameters in the uri, but this is not one.]

Reply:
This one was initially boolean and from a previous ARM reviewer we were asked to change the state to enum instead.

Reply:
Yes, but enum values of 'True' and 'False' are really no better than a boolean. You should either change it to a proper enum with values that mean something on their own, so you can really extend it in a meaningful way, or just use a boolean, if you're sure you never need to extend it.